app/[locale]/job/add/page.tsx

```diff
@@ -2,29 +2,32 @@
 import { Grid, GridContainer } from '@trussworks/react-uswds'
 import { useTranslation } from 'react-i18next'
 import { useAppDispatch, useAppSelector } from "@/lib/hooks"
-import { addJob, JobItem, selectIncomeItems } from "@/lib/features/job/jobSlice"
+import { addJob, SetJobPayload, selectJobCount, JobItem } from "@/lib/features/job/jobSlice"
 import { useRouter } from "next/navigation"
 import VerifyNav from "@/app/components/VerifyNav"
 import IncomeFormJob, { IncomeFormJobData } from '@/app/[locale]/job/IncomeFormJob'
+import { createUuid } from '@/lib/store'
 
 export default function Page() {
     const { t } = useTranslation()
     const dispatch = useAppDispatch()
     const router = useRouter()
-    const items = useAppSelector(state => selectIncomeItems(state))
-    const jobCount = items.length
+    const jobCount = useAppSelector(state => selectJobCount(state))
 
-    function addIncomeClicked({description, business, taxesFiled, payments=[]}: IncomeFormJobData) {
-
-        const jobItem: JobItem = {
-            description,
-            business,
-            taxesFiled,
-            payments
+    function addIncomeClicked({description, business, taxesFiled}: IncomeFormJobData) {
```

Comment:
you'll see the paradigm of `JobItem` and `SetJobPayload` across all three. the point here was to be able to submit the id of the object to the slice for adding the item, but keep that id out of the `JobItem`, since the id is stored as the key to object and not on it directly.

structure, for reference: 
```
{
  byId: {
    [id: string]: JobItem
  },
  [allIds: Array<string>]: [id]
}
```